odl/test/tomo/data/mrc2014_test.py

```diff
@@ -0,0 +1,243 @@
+
+from itertools import permutations
+import pytest
+import numpy as np
+import tempfile
+
+from odl.tomo.data import mrc_header_from_params, FileWriterMRC, FileReaderMRC
+from odl.util.testutils import all_equal
+
+
+# --- pytest fixtures --- #
+
+
+mode_dtype_params = [(0, 'int8'), (1, 'int16'), (2, 'float32'), (6, 'uint16')]
+mode_dtype_ids = [" mode = {p[0]}, dtype = '{p[1]}' ".format(p=p)
+                  for p in mode_dtype_params]
+
+
+@pytest.fixture(scope='module', ids=mode_dtype_ids, params=mode_dtype_params)
+def mrc_mode_dtype(request):
+    mode, dtype = request.param
+    return mode, np.dtype(dtype)
+
+
+axis_order_params = list(permutations((0, 1, 2)))
+axis_order_ids = [' axis_order = {} '.format(p) for p in axis_order_params]
+
+
+@pytest.fixture(scope='module', ids=axis_order_ids, params=axis_order_params)
+def axis_order(request):
+    return request.param
+
+
+ncells_params = [(5, 10, 20), (1, 5, 6), (10, 1, 1), (1, 1, 1)]
+ncells_ids = [' ncells = {} '.format(p) for p in ncells_params]
+
+
+@pytest.fixture(scope='module', ids=ncells_ids, params=ncells_params)
+def ncells(request):
+    return request.param
+
+
+ispg_kind_params = [(0, 'projections'), (1, 'volume')]
+ispg_kind_ids = [" ispg = {p[0]}, kind = '{p[0]}' ".format(p=p)
+                 for p in ispg_kind_params]
+
+
+@pytest.fixture(scope='module', ids=ispg_kind_ids, params=ispg_kind_params)
+def ispg_kind(request):
+    return request.param
+
+
+def test_mrc_header_from_params_defaults(ncells, mrc_mode_dtype, ispg_kind):
+    """Test the utility function for the minimal required parameters."""
+    mode, dtype = mrc_mode_dtype
+    true_ispg, kind = ispg_kind
+    header = mrc_header_from_params(ncells, dtype, kind)
+
+    # Check values of all header entries
+    nx = header['nx']['value']
+    ny = header['ny']['value']
+    nz = header['nz']['value']
+    assert all_equal([nx, ny, nz], ncells)
+    assert header['mode']['value'] == mode
+    mx = header['mx']['value']
+    my = header['my']['value']
+    mz = header['mz']['value']
+    assert all_equal([mx, my, mz], ncells)
+    cella = header['cella']['value']
+    assert all_equal(cella, np.ones(3) * ncells)
+    mapc = header['mapc']['value']
+    mapr = header['mapr']['value']
+    maps = header['maps']['value']
+    assert all_equal([mapc, mapr, maps], [1, 2, 3])
+    dmin = header['dmin']['value']
+    dmax = header['dmax']['value']
+    dmean = header['dmean']['value']
+    rms = header['rms']['value']
+    assert all_equal([dmin, dmax, dmean, rms], [1.0, 0.0, -1.0, -1.0])
+    ispg = header['ispg']['value']
+    assert ispg == true_ispg
+    nsymbt = header['nsymbt']['value']
+    assert nsymbt == 0
+    exttype = header['exttype']['value']
+    assert np.array_equal(exttype, np.fromiter('    ', dtype='S1'))
+    nversion = header['nversion']['value']
+    assert nversion == 20140
+    origin = header['origin']['value']
+    assert np.array_equal(origin, np.zeros(3, dtype=int))
```

Comment:
`label` is a `(10, 80)` array of empty single byte strings, so your suggestions doesn't work. Also the `all_equal` thing doesn't work well with Numpy arrays that are not 1D.